bruno/io-lollipop/firstLCSignedMessage.bru

```diff
@@ -0,0 +1,97 @@
+meta {
+  name: firstLCSignedMessage
+  type: http
+  seq: 10
+}
+
+post {
+  url: {{IO_LOLLIPOP_BASE_URL}}{{IO_LOLLIPOP_API_PATH}}/first-lollipop-consumer/signed-message
+  body: json
+  auth: none
+}
+
+headers {
+  ~x-functions-key: SET_IN_PREREQUEST_SCRIPT
+}
+
+body:json {
+  { "message": "hello world" }
+}
+
+body:text {
+  {"message":"hello"}
+}
+
+script:pre-request {
+  const crypto = require("node:crypto");
+  const jose = require("node-jose");
+  const mattrglobal = require("@mattrglobal/http-signatures");
+  
+  // Set header 'x-functions-key' if the necessary key is found in the Environment
+  const ioLollipopApiKey = bru.getEnvVar("IO_LOLLIPOP_API_KEY");
+  if (ioLollipopApiKey != null ) {
+    req.setHeader("x-functions-key", ioLollipopApiKey);
+  }
+  
+  const lollipopAssertionRef = bru.getVar("lollipopAssertionRef");
+  if (!lollipopAssertionRef) {
+    throw new Error("Var 'lollipopAssertionRef' is not defined! Run 'reservePubKey' and 'activatePubKey' before 'generateLCParams'");
+  }
+  
+  const lollipopThumbprint = bru.getVar("lollipopThumbprint");
+  if (!lollipopThumbprint) {
+    throw new Error("Var 'lollipopThumbprint' is not defined! Run 'reservePubKey' and 'activatePubKey' before calling API");
+  }
+  
+  const method = req.method;
+  const url = `https://localhost:8000/first-lollipop/sign`;
+  const alg = mattrglobal.AlgorithmTypes["ecdsa-p256-sha256"];
+  const keyid = lollipopThumbprint;
+  const privateKey = JSON.parse(Buffer.from(jose.util.base64url.decode(bru.getVar("encodedPrivateKey"))));
```

Comment:
can you add a specific variable for the encodedPrivateKey along with a dedicated throw?

Reply:
you're right